packages/eas-cli/src/commands/env/push.ts

```diff
@@ -0,0 +1,171 @@
+
+    if (!environment) {
+      throw new Error('Please provide an environment to push the env file to.');
+    }
+
+    const updateVariables: Record<string, EnvironmentVariablePushInput> =
+      await this.parseEnvFileAsync(envPath, environment);
+
+    const variableNames = Object.keys(updateVariables);
+
+    const { appVariables: existingVariables } = await EnvironmentVariablesQuery.byAppIdAsync(
+      graphqlClient,
+      { appId: projectId, environment, filterNames: variableNames }
+    );
+
+    // Check if any of the variables already exist in the environment. Prompt the user to overwrite them.
+    const existingSensitiveVariables = existingVariables.filter(
+      variable => variable.value === null
+    );
+
+    if (existingSensitiveVariables.length > 0) {
+      const existingSensitiveVariablesNames = existingSensitiveVariables.map(
+        variable => variable.name
+      );
+      Log.warn('Sensitive variables cannot be overwritten with the `push` command.');
+      Log.warn('Use `eas env:update` to update them:');
+      existingSensitiveVariablesNames.forEach(name => Log.warn(`- ${name}`));
+    }
+
+    const existingDifferentVariables: EnvironmentVariableFragment[] = [];
+    // Remove variables that are the same as the ones in the environment or sensitive.
+    existingVariables.forEach(variable => {
+      const existingVariableUpdate = updateVariables[variable.name];
+      if (
+        existingVariableUpdate &&
+        existingVariableUpdate.value !== variable.value &&
+        variable.value !== null
+      ) {
+        existingDifferentVariables.push(variable);
+      } else {
+        delete updateVariables[variable.name];
+      }
+    });
+
+    const existingDifferentSharedVariables = existingDifferentVariables.filter(
+      variable => variable.scope === EnvironmentVariableScope.Shared
+    );
+
+    if (existingDifferentSharedVariables.length > 0) {
+      const existingDifferentSharedVariablesNames = existingDifferentSharedVariables.map(
+        variable => variable.name
+      );
+      Log.error('Shared variables cannot be overwritten.');
+      Log.error('Remove them from the env file or unlink them from the project to continue:');
+      existingDifferentSharedVariablesNames.forEach(name => Log.error(`- ${name}`));
+      throw new Error('Shared variables cannot be overwritten by eas env:push command');
+    }
+
+    if (existingDifferentVariables.length > 0) {
+      Log.warn('Some variables already exist in the environment.');
+
+      const { variablesToOverwrite } = await promptAsync({
+        type: 'multiselect',
+        name: 'variablesToOverwrite',
+        message: 'Select variables to overwrite:',
+        // @ts-expect-error property missing from `@types/prompts`
+        optionsPerPage: 20,
+        choices: existingDifferentVariables.map(variable => ({
+          title: `${variable.name}: ${updateVariables[variable.name].value}`,
+          value: variable.name,
+        })),
+      });
```

Comment:
I think that if one wants to push 50 env vars, this can be annoying. Maybe we want to first ask users
```
The ENV_VAR1, ENV_VAR2, ENV_VAR3 environment variables already exist in ${environment} environment. Do you want to override them all?
```
and if not fallback to multi-select.